### PR TITLE
#838 - Message Passing Skeleton

### PR DESCRIPTION
**Pull request by gisjedi**
This pull request implements the functionality outlined in #838 for initial MPI support within Scale. Documentation is included for how to set up SQS and RabbitMQ.

### CODE REVIEW

File: scale/messaging/manager.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        """Serialize CommandMessage and send via configured message broker
+
+        The command.to_json() and command.message_type will be used to generate
+        serialized form of CommandMessage for transmission across the wire.
+
+        :param command: CommandMessage to be sent via configured broker
+        :type command: `messaging.messages.message.CommandMessage`
+        """
+
+        self._backend.send_message({"type": command.message_type, "body": command.to_json()})
```
Comment (dfaller) [inline scale/messaging/manager.py]: Can we not send multiple messages at a time?

File: scale/docs/architecture/index.rst [hunk 2]
```diff
@@ -11,6 +11,7 @@ Architecture
    overview
    jobs/index
    logs
+   messaging
```
Comment (dfaller) [inline scale/docs/architecture/index.rst]: This new app needs to be added to appropriate index.rst doc file for generated code docs.

File: scale/messaging/backends/amqp.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import Queue
+import logging
+from contextlib import closing
+
+from kombu import Connection
+
+from messaging.backends.backend import MessagingBackend
```
Comment (gisjedi) [inline scale/messaging/backends/amqp.py]: Discussed as Queue being an actual stdlib import.

File: scale/docs/architecture/index.rst (hunk 2 above)
Comment (gisjedi) [inline scale/docs/architecture/index.rst]: I believe this is included as it is chained from the root index.rst -> architecture.rst -> messaging.rst

File: scale/messaging/backends/sqs.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+logger = logging.getLogger(__name__)
+
+
+class SQSMessagingBackend(MessagingBackend):
+    """Backend supporting message passing via Amazon SQS"""
+
+    def __init__(self):
+        super(SQSMessagingBackend, self).__init__('sqs')
+
+        self._region_name = self._broker.get_broker()
```
Comment (gisjedi) [inline scale/messaging/backends/sqs.py]: This is a bit hokey looking, but correct. Maybe I'll change the name of the getter to be clearer.

File: scale/messaging/manager.py [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+        messages = self._backend.receive_messages(10)
+
+        for message in messages:
+            try:
+                self._process_message(message)
+            except InvalidCommandMessage:
+                logger.exception('Exception encountered processing message payload.')
+            except CommandMessageExecuteFailure:
+                logger.exception('CommandMessage failure during execute call.')
```
Comment (dfaller) [inline scale/messaging/manager.py]: How do the messages get deleted/acked?

**Pull request merged**